### ISSUE TITLE
Deprecate `execute` and `refresh` for a unified name like `run`

### ISSUE THREAD

**Issue by TheAlexLichter** [nuxt/nuxt#30049]
### Describe the feature

Right now, `useAsyncData` and `useFetch` expose a `refresh`/`execute` method which does the same but is exported under these two names for semantic reasons.

IMO it would be better to have a simplified `run` functionality which could be called in either semantic (or being renamed as needed). This could avoid confusion, as I see often that people aren't aware the current two exports are the same.

### Additional information

- [x] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [x] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [x] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by stephenjason89** [comment]
May I help with this issue?

**Comment by manniL** [comment]
@stephenjason89 We still have to decide on the names and whether we will deprecate at all. Maybe [another good first issue](https://github.com/nuxt/nuxt/issues?q=sort%3Aupdated-desc+is%3Aopen+label%3A%22good+first+issue%22) would be better and straight away actionable! 👍

**Comment by cjpearson** [comment]
I agree that having just one function would simplify things. Probably I would lean towards `execute`. Language-wise it works both for requests or functions and unlike `refresh` it doesn't imply that the data was already fetched. It also has the benefit of already existing in Nuxt3. 

If we want to go with a new name, maybe `fetch`?

**Comment by vacijj** [comment]
+1 for simplification. Having only one function definitely decreases complexity (especially for new users). 👍 

I'd like to add some more function names to the list.

| Function  | Rating            | Thoughts                                                                                                                               |
|:----------|:-----------------:|:---------------------------------------------------------------------------------------------------------------------------------------|
| `refresh` | -                 | Implies that the purpose is to refresh data, indicating that the initial data load has already taken place.                            |
| `execute` | -                 | Describes the execution of an action/command rather than loading data.                                                                 |
| `run`     | -                 | Kind of similar to `execute`. Describes the run of an action/command rather than loading data.                                         |
| `fetch`   | -                 | Conflicting with native `fetch` and `$fetch`.                                                                                          |
| `load`    | :1st_place_medal: | At the moment, my personal favorite. Descriptive and understandable. Matching for both `useAsyncData` and `useFetch`. Not conflicting. |
| `reload`  | -                 | Similar to `refresh`. Implies that the purpose is to reload data, indicating that the initial data load has already taken place.       |
| `get`     | -                 | Conflicting or restricted to the http method `GET`. Too generic.                                                                       |
| `query`   | :3rd_place_medal: | Ok, but conflicting with [`UseFetchOptions.query` param](https://nuxt.com/docs/api/composables/use-fetch#params).                      |
| `request` | :2nd_place_medal: | Ok, but also conflicting with the commonly used `request` in HTTP context, ...                                                         |

@manniL @danielroe I hope it helps in your [Team Board Discussions](https://github.com/orgs/nuxt/projects/8/views/1).

**Comment by Amar-Gill** [comment]
I think `run` makes sense semantically. Another advantage is that it's only 3 keys to type.

If you want to map it to another name you can always do `const { run: refresh } = useFetch()`. Although a few extra keys to type.

Yes I know I'm lazy =]

**Comment by Cyanhall** [comment]
I recommend naming it `fetchData`, as it intuitively conveys its purpose.

**Comment by lessainspasorchies** [comment]
I don't understand what's the deal with switching away from `refresh`. These are data fetching composables, let's be explicit. I was more than OK with `refresh`, and I could understand `execute` as _execute the composable again_ even though it's less clear in my purely personal preference. Why not `refetchData()`?
IMHO, everyone will have a different opinion based on its preferences. I'm more than OK with narrowing down to 1 unified name as long as it states clearly what its purpose is. I've always loved Nuxt for that clear, explicit terminology, let's make sure it stays like this.
Thank you for your consideration!

**Comment by TheAlexLichter** [comment]
As you can also decide to *not* run the call immediately (via `immediate: false`), either `refresh` or `execute` would be a fitting name - but only for one of the scenarios, **but not both**.
To unify it, people are also not confused about two exports of the exact same function.

So indeed, we try to make things more clear 😋

**Comment by TheAlexLichter** [comment]
After the discussion & opinions here + giving this more thoughts internally, changing the name to a *single* name would cause more issues (either too general or not capturing both purposes).

Instead, adaptions to the JSDocs will be made so that it is clear that `refresh` and `execute` are the same export!

Thanks to everyone for your input 🙏🏻
